homeassistant/components/refoss/switch.py

```diff
@@ -0,0 +1,74 @@
+"""Switch for Refoss."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from refoss_ha.controller.toggle import ToggleXMix
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import COORDINATORS, DISPATCH_DEVICE_DISCOVERED, DISPATCHERS, DOMAIN
+from .entity import RefossEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Refoss device from a config entry."""
+
+    @callback
+    def init_device(coordinator):
+        """Register the device."""
+        device = coordinator.device
+        if not isinstance(device, ToggleXMix):
+            return
+
+        new_entities = []
+        for channel in device.channels:
+            entity = RefossSwitch(coordinator, channel=channel)
+            new_entities.append(entity)
+
+        async_add_entities(new_entities)
+
+    for coordinator in hass.data[DOMAIN][COORDINATORS]:
+        init_device(coordinator)
+
+    hass.data[DOMAIN][DISPATCHERS].append(
+        async_dispatcher_connect(hass, DISPATCH_DEVICE_DISCOVERED, init_device)
+    )
+
+
+class RefossSwitch(RefossEntity, SwitchEntity):
+    """Entity that controls switch based refoss device."""
+
+    def __init__(self, coordinator, channel: int) -> None:
+        """Construct."""
+        super().__init__(coordinator, channel)
+        self._channel_id = channel
```

Comment:
Are other platforms also going to use channel_id? If so, you could set it in the RefossEntity

Reply:
yes，ok